azure-maven-plugin-lib/src/main/java/com/microsoft/azure/maven/auth/AzureClientFactory.java

```diff
@@ -31,22 +31,17 @@
     public static Azure getAzureClient(AzureTokenWrapper azureTokenCredentials, String subscriptionId,
                                        String userAgent) throws IOException, AzureLoginFailureException {
         Preconditions.checkNotNull(azureTokenCredentials, "The parameter 'azureTokenCredentials' cannot be null.");
-        Log.info(azureTokenCredentials.getCredentialDescription());
         final Authenticated authenticated = Azure.configure().withUserAgent(userAgent).authenticate(azureTokenCredentials);
         // For cloud shell, use subscription in profile as the default subscription.
         if (StringUtils.isEmpty(subscriptionId) && AzureAuthHelperLegacy.isInCloudShell()) {
             subscriptionId = AzureAuthHelperLegacy.getSubscriptionOfCloudShell();
         }
         subscriptionId = StringUtils.isEmpty(subscriptionId) ? azureTokenCredentials.defaultSubscriptionId() : subscriptionId;
-        final Azure azureClient = StringUtils.isEmpty(subscriptionId) ? authenticated.withDefaultSubscription() :
+        return StringUtils.isEmpty(subscriptionId) ? authenticated.withDefaultSubscription() :
                 authenticated.withSubscription(subscriptionId);
-        checkSubscription(azureClient, subscriptionId);
-        final Subscription subscription = azureClient.getCurrentSubscription();
-        Log.info(String.format(SUBSCRIPTION_TEMPLATE, subscription.displayName(), subscription.subscriptionId()));
-        return azureClient;
     }
 
-    private static void checkSubscription(Azure azure, String targetSubscription) throws AzureLoginFailureException {
+    public static void checkSubscription(Azure azure, String targetSubscription) throws AzureLoginFailureException {
```

Comment:
If we separate `checkSubscription` from `getAzureClient`, do we need to add  `checkSubscription` back to `getAzureClient` caller

Reply:
I will roll this change back